### PR TITLE
Try get port operational speed from STATE DB

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
HLD change: https://github.com/Azure/SONiC/pull/932

**What I did**

1. Put port operational speed to STATE DB
2. Remove previous workaround https://github.com/Azure/sonic-swss/pull/1769

**Why I did it**

For issue https://github.com/Azure/sonic-buildimage/issues/9595

**How I verified it**

Manual test

**Details if related**

### CODE REVIEW

File: portsyncd/portsyncd.cpp [hunk 1]
```diff
@@ -247,54 +242,7 @@ void handlePortConfig(ProducerStateTable &p, map<string, KeyOpFieldsValuesTuple>
             /* No support for port delete yet */
             if (op == SET_COMMAND)
             {
-                
-                for (auto i : values)
```
Comment (prsunny) [inline portsyncd/portsyncd.cpp:251]: This change is not relevant to the PR description. Please revert

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -5609,9 +5609,13 @@ void PortsOrch::doTask(NotificationConsumer &consumer)
                 if (getPortOperSpeed(port, speed))
                 {
                     SWSS_LOG_NOTICE("%s oper speed is %d", port.m_alias.c_str(), speed);
-                    updateDbPortOperSpeed(port, speed);
+                    updatePortOperSpeedToStateDB(port, speed);
```
Comment (prsunny) [inline orchagent/portsorch.cpp:5612]: I think we should keep the API as is and update to both APP and STATE DB

**Review by prsunny — CHANGES_REQUESTED** [review]

File: portsyncd/portsyncd.cpp (hunk 1 above)
Comment (Junchao-Mellanox) [reply portsyncd/portsyncd.cpp:251]: This change is related to description:

2. Remove previous workaround https://github.com/Azure/sonic-swss/pull/1769

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (Junchao-Mellanox) [reply orchagent/portsorch.cpp:5612]: Per our discussion, we will put the speed to STATE DB currently. I will rename the API back to previous.

File: portsyncd/portsyncd.cpp (hunk 1 above)
Comment (prgeor) [reply portsyncd/portsyncd.cpp:251]: @prsunny we discussed this offline which was a hacky code put earlier as mentioned by junchao. so this code can be removed

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -5612,6 +5612,10 @@ void PortsOrch::doTask(NotificationConsumer &consumer)
                     updateDbPortOperSpeed(port, speed);
                 }
             }
+            else
+            {
+                updateDbPortOperSpeed(port, 0);
+            }
```
Comment (prgeor) [inline orchagent/portsorch.cpp:5618]: i think we should update speed=0 only when the operational status is UP similar to done in the failure case [here](https://github.com/Azure/sonic-swss/pull/2119/files#diff-bb736c6a11da68afacf95d761fb52a3c11726cdd72c967d4803b8c9b1a6dcb69R5728)

**Review by prsunny — COMMENTED** [review]
lgtm, @prgeor to approve/merge

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@Junchao-Mellanox can you check the LGTM failure

File: orchagent/portsorch.cpp [hunk 4]
```diff
@@ -5611,8 +5611,12 @@ void PortsOrch::doTask(NotificationConsumer &consumer)
                     SWSS_LOG_NOTICE("%s oper speed is %d", port.m_alias.c_str(), speed);
                     updateDbPortOperSpeed(port, speed);
                 }
+                else
+                {
+                    updateDbPortOperSpeed(port, 0);
+                }
             }
-
+
```
Comment (prsunny) [inline orchagent/portsorch.cpp:5619]: Is this tab space?

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
> @Junchao-Mellanox can you check the LGTM failure

@prgeor , this is a known issue

**Pull request merged**